File: src/Entity/Events.php
<?php

namespace App\Entity;

use App\Repository\EventsRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity(repositoryClass: EventsRepository::class)]
class Events
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message: "Veuillez ajouter un nom ")]
    private ?string $name = null;

    #[ORM\Column(type: Types::DECIMAL, precision: 10, scale: 2)]
    #[Assert\NotBlank(message: "Veuillez ajouter un prix.")]
    #[Assert\Regex(
        pattern: "/^\d+(\.\d{1,2})?$/",
        message: "Le prix doit être un nombre valide avec jusqu'à deux décimales."
    )]
    private ?string $price = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message: "Veuillez ajouter un état.")]
    private ?string $state = null;

    #[ORM\Column(length: 255)]
    private ?string $img = null;

    #[ORM\Column]
    #[Assert\NotBlank(message: "Veuillez ajouter une date.")]
    private ?\DateInterval $date = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message: "Veuillez ajouter une description.")]
    private ?string $description = null;

    #[ORM\OneToMany(targetEntity: Activities::class, mappedBy: 'events')]
    #[Assert\NotBlank(message: "Veuillez ajouter une activite.")]
    private Collection $Activities;

    #[ORM\ManyToMany(targetEntity: Activities::class, inversedBy: 'event')]
    private Collection $events;

    public function __construct()
    {
        $this->Activities = new ArrayCollection();
        $this->events = new ArrayCollection();
    }

    

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getName(): ?string
    {
        return $this->name;
    }

    public function setName(string $name): static
    {
        $this->name = $name;

        return $this;
    }

    public function getPrice(): ?string
    {
        return $this->price;
    }

    public function setPrice(string $price): static
    {
        $this->price = $price;

        return $this;
    }

    public function getState(): ?string
    {
        return $this->state;
    }

    public function setState(string $state): static
    {
        $this->state = $state;

        return $this;
    }

    public function getImg(): ?string
    {
        return $this->img;
    }

    public function setImg(string $img): static
    {
        $this->img = $img;

        return $this;
    }

    public function getDate(): ?\DateInterval
    {
        return $this->date;
    }

    public function setDate(\DateInterval $date): static
    {
        $this->date = $date;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(string $description): static
    {
        $this->description = $description;

        return $this;
    }

    /**
     * @return Collection<int, Activities>
     */
    public function getActivities(): Collection
    {
        return $this->Activities;
    }

    public function addActivity(Activities $activity): static
    {
        if (!$this->Activities->contains($activity)) {
            $this->Activities->add($activity);
            $activity->setEvents($this);
        }

        return $this;
    }

    public function removeActivity(Activities $activity): static
    {
        if ($this->Activities->removeElement($activity)) {
            // set the owning side to null (unless already changed)
            if ($activity->getEvents() === $this) {
                $activity->setEvents(null);
            }
        }

        return $this;
    }

    /**
     * @return Collection<int, Activities>
     */
    public function getEvents(): Collection
    {
        return $this->events;
    }

    public function addEvent(Activities $event): static
    {
        if (!$this->events->contains($event)) {
            $this->events->add($event);
        }

        return $this;
    }

    public function removeEvent(Activities $event): static
    {
        $this->events->removeElement($event);

        return $this;
    }
}

File: src/main/java/tn/esprit/models/Events.java
package tn.esprit.models;

import java.text.SimpleDateFormat;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

public class Events {

    private int id;
    private String name, state, description, img;
    private float price;
    private Date date;
    private List<Activities> activities;

    public Events(String name, String state, String description, String img, float price, Date date) {
        this.name = name;
        this.state = state;
        this.description = description;
        this.img = img;
        this.price = price;
        this.date = date;
    }

    public Events(int id, String name, String state, String description, String img, float price, Date date) {
        this.id = id;
        this.name = name;
        this.state = state;
        this.description = description;
        this.img = img;
        this.price = price;
        this.date = date;
    }

    public Events() {
    }

    public Events(String name, String state, String description, String img, float price) {

        this.name = name;
        this.state = state;
        this.description = description;
        this.img = img;
        this.price = price;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getState() {
        return state;
    }

    public void setState(String state) {
        this.state = state;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getImg() {
        return img;
    }

    public void setImg(String img) {
        this.img = img;
    }

    public float getPrice() {
        return price;
    }

    public void setPrice(float price) {
        this.price = price;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }

    public List<Activities> getActivities() {
        return activities;
    }

    public void setActivities(List<Activities> activities) {
        this.activities = activities;
    }

    public List<String> getActivityNames() {
        List<String> activityNames = new ArrayList<>();
        if (activities != null) {
            for (Activities activity : activities) {
                activityNames.add(activity.getName());
            }
        }
        return activityNames;
    }



    public void addActivity(Activities activity) {
        activities.add(activity);
    }

    @Override
    public String toString() {
        return "Events{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", state='" + state + '\'' +
                ", description='" + description + '\'' +
                ", img='" + img + '\'' +
                ", price=" + price +
                ", date=" + date +
                '}';
    }

    // Format date as per Symfony's DateInterval format
    private String formatDate(Date date) {
        SimpleDateFormat formatter = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss");
        return formatter.format(date);
    }
}
